src/sagemaker/session.py

```diff
@@ -3593,6 +3643,19 @@ def create_model_package_from_containers(
                 "IMAGE_CLASSIFICATION", "OBJECT_DETECTION", "TEXT_GENERATION", "IMAGE_SEGMENTATION",
                 "CLASSIFICATION", "REGRESSION", "OTHER" (default: None).
         """
+        if containers:
+            # Containers are is provided. Now we can merge missing entries from config.
+            # If Containers are not provided, it is safe to ignore. This is because,
+            # if this object is provided to the API, then Image is required for Containers.
+            # That is not supported by the config now. So if we merge values from config,
+            # then API will throw an exception. In the future, when SageMaker Config starts
+            # supporting other parameters we can add that.
+            update_list_of_dicts_with_values_from_config(
+                containers,
+                MODEL_PACKAGE_INFERENCE_SPECIFICATION_CONTAINERS_ENVIRONMENT_PATH,
+                required_key_paths=["Image"],
+                sagemaker_session=self,
+            )
```

Comment:
Thanks for being careful about these cases. Wouldnt the behavior be the same if `if containers:` is removed? I think the utils helper should cover this case